beancount_import/source/schwab_csv.py

```diff
@@ -307,6 +321,38 @@ def get_processed_entry(
                 fees=self.fees,
                 **shared_attrs,
             )
+        if self.action is BrokerageAction.EXPIRED:
+            assert self.quantity is not None
+            amount = Amount(Decimal(0), currency=CASH_CURRENCY) if self.amount is None else self.amount
+            price = Decimal(0) if self.price is None else self.price
+            shared_attrs: SharedAttrsDict = dict(
+                capital_gains_account=capital_gains_account,
+                fees_account=fees_account,
+                account=account,
+                date=self.date,
+                action=self.action,
+                description=self.description,
+                quantity=self.quantity,
+                symbol=self.symbol,
+                price=price,
+                fees=self.fees,
+                amount=amount,
+                filename=self.filename,
+                line=self.line,
+            )
+            # an expiring long option means it is sold at the end => the posting has a negative 'quantity'
+            return Buy(**shared_attrs) if self.quantity > 0 else Sell(**shared_attrs)
+
+        if self.action in (BrokerageAction.SHORT_TERM_CAP_GAIN, BrokerageAction.LONG_TERM_CAP_GAIN):
+            return FundGainsDistribution(symbol=self.symbol, capital_gains_account=capital_gains_account, **shared_attrs)
+
+        if self.action == BrokerageAction.STOCK_MERGER:
+            assert self.quantity is not None
+            return Merger(account=account, fees_account=fees_account, symbol=self.symbol, quantity=self.quantity, price=self.price,
+                          amount=self.amount, merger_symbol=self.merger_symbol, merger_quantity=self.merger_quantity,
+                          date=self.date, action=self.action, description=self.description, fees=self.fees, filename=self.filename,
+                          merger_description=self.merger_description, line=self.line)
```

Comment:
again here it looks like we are re-handling things that are already handled in `shared_attrs`, and then not using `shared_attrs`. This makes future maintenance more difficulty; if we need to make global adjustments that apply to a certain shared attribute of all transaction types, it will appear that we can make this change in `shared_attrs`, and we may not notice that some transaction types aren't using it. So I'd rather use the existing pattern of `STOCK_PLAN_ACTIVITY` as mentioned above to ensure that all transaction types actually do share `shared_attrs`. If we have to use that pattern too many times for some field, we can move that field out of `shared_attrs` and instead pass it individually for each transaction type.

Reply:
Done.